### ISSUE TITLE
MA1096 (inheritdoc on non-override) triggers for inheritdoc on type

### ISSUE THREAD

**Issue by Timovzl** [meziantou/Meziantou.Analyzer#1139]
### Version of the Meziantou.Analyzer NuGet package

3.0.74

### Rule Identifier

MA0196

### Target Framework

net10.0

### C# Language version

C# 14

### Description

MA0196, "Do not use inheritdoc on non-inheriting members", sometimes triggers on instances of inheritdoc on types. I noticed this after suppressing MA0197 and still seeing a few warnings on inheritdoc on types.

The one thing I could find that these cases have in common is that _either_ they're generic types _or_ they have both a base type plus an interface. Hopefully that helps find the issue!

P.S. Respectfully, I find MA0197 to be unusually opinionated as compared to all its siblings. There is a very strong case for inheritdoc on types, namely every time things appear in pairs like this: interface plus implementation; interface plus abstract class; non-generic plus generic class; etc. It is extremely valid to both (A) want the documentation to pop up for either type, and (B) want to avoid maintaining duplicated documentation. In fact, most teams have "interface plus implementation" for each of their dependencies. As such, I would recommend completely disabling MA0197 by default - not a suggestion, not silent, but off.

### Reproduction Steps

```c#
/// <inheritdoc/>
public abstract class SqliteUnitTest() : SqliteTestBase<TestDbContext>(), IUnitTest
{
}
```

**Comment by meziantou** [comment]
I think it reports the primary constructor. Indeed, the XML comment applies to the class and the constructor (member). I will skip this case as the comment mostly apply to the class.

**Comment by meziantou** [comment]
For MA0197, can you share examples where you think the rule should not report diagnostics (would be better in another issue).
I know this is opinionated, but I can't find a lot of valid cases for it. Also, enabling the rule by default is the only way to get feedbacks from users.
Feel free to lower severity or disable the rule in the editorconfig meanwhile.
